### ISSUE TITLE
kiwi-ng boxbuild "fails" without any trace of info on "unsupported" distribution

### ISSUE THREAD

**Issue by jirib** [OSInside/kiwi#1770]
## Problem description

I was trying to test `kiwi-ng` with `boxbuild` plugin (installed via `pip`) on Void Linux. Unfortunatelly it ends and prints *help* output and there's no trace of comprehensible info why it does not work. BTW I was using a command from official docs.

## Expected behaviour

At least some info about why the command does not work.

## Steps to reproduce the behaviour

Use some distro which is not suse, fedora, centos, rhel, debian/ubuntu, I used Void Linux.

## OS and Software information
+ KIWI version: 

`kiwi==9.23.20`

+ Operating system: 

`Void 5.10.23_1 x86_64 GenuineIntel notuptodate rrrdFFFFFFF`

+ OBS version: 

???

```
kiwi-ng --type iso system boxbuild --box leap --description suse-leap-15.2 --target-dir /h
ome/jiri/tmp/myimage
> /home/jiri/.venv/bin/kiwi-ng(33)<module>()                                                                          
-> sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])          
(Pdb) ll      
  1     #!/home/jiri/.venv/bin/python                   
  2     # EASY-INSTALL-ENTRY-SCRIPT: 'kiwi==9.23.20','console_scripts','kiwi-ng'
  3     import re                                                                                                     
  4     import sys                                                                                                    
  5     
  6     # for compatibility with easy_install; see #2198   
  7     __requires__ = 'kiwi==9.23.20'                                                                                
  8                     
  9     try:                                                                                                          
 10         from importlib.metadata import distribution                                                               
 11     except ImportError:                                
 12         try:                                           
 13             from importlib_metadata import distribution 
 14         except ImportError:           
 15             from pkg_resources import load_entry_point
 16                                                        
 17                     
 18     def importlib_load_entry_point(spec, group, name):                                                            
 19         dist_name, _, _ = spec.partition('==')
 20         matches = (                                    
 21             entry_point
 22             for entry_point in distribution(dist_name).entry_points
 23             if entry_point.group == group and entry_point.name == name
 24         )                                              
 25         return next(matches).load()
 26                                                        
 27                                                                                                                   
 28     globals().setdefault('load_entry_point', importlib_load_entry_point)                                          
 29                          
 30                                                        
 31     if __name__ == '__main__':
 32         breakpoint()                                   
 33  ->     sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])                                         
 34         sys.exit(load_entry_point('kiwi==9.23.20', 'console_scripts', 'kiwi-ng')())
(Pdb) sys.argv[0]                                          
'/home/jiri/.venv/bin/kiwi-ng'
(Pdb) n                 
> /home/jiri/.venv/bin/kiwi-ng(34)<module>()                                                                          
-> sys.exit(load_entry_point('kiwi==9.23.20', 'console_scripts', 'kiwi-ng')())                                        
(Pdb) ll                    
  1     #!/home/jiri/.venv/bin/python
  2     # EASY-INSTALL-ENTRY-SCRIPT: 'kiwi==9.23.20','console_scripts','kiwi-ng'
  3     import re
  4     import sys
  5  
  6     # for compatibility with easy_install; see #2198
  7     __requires__ = 'kiwi==9.23.20'
  8  
  9     try:
 10         from importlib.metadata import distribution
 11     except ImportError:
 12         try:
 13             from importlib_metadata import distribution 
 14         except ImportError:
 15             from pkg_resources import load_entry_point
 16  
 17  
 18     def importlib_load_entry_point(spec, group, name):
 19         dist_name, _, _ = spec.partition('==')
 20         matches = (
 21             entry_point
 22             for entry_point in distribution(dist_name).entry_points
 23             if entry_point.group == group and entry_point.name == name
 24         )
 25         return next(matches).load()
 26  
 27  
 28     globals().setdefault('load_entry_point', importlib_load_entry_point)
 29  
 30  
 31     if __name__ == '__main__':
 32         breakpoint()
 33         sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
 34  ->     sys.exit(load_entry_point('kiwi==9.23.20', 'console_scripts', 'kiwi-ng')())
(Pdb) n
usage: kiwi-ng [global options] service <command> [<args>]

       kiwi-ng system boxbuild -h | --help
       kiwi-ng system boxbuild --box=<name>
           [--box-memory=<vmgb>]
           [--box-debug]
           [--kiwi-version=<version>]
           [--shared-path=<path>]
           [--no-update-check]
           [--no-snapshot]
           [--9p-sharing | --virtiofs-sharing]
           [--x86_64 | --aarch64]
           [--machine=<qemu_machine>]
           [--cpu=<qemu_cpu>] 
           <kiwi_build_command_args>...
       kiwi-ng system boxbuild --list-boxes
       kiwi-ng system boxbuild help

global options:
    --color-output
        use colors for warning and error messages
    --config=<configfile>
        use specified runtime configuration file. If
        not specified the runtime configuration is looked
        up at ~/.config/kiwi/config.yml or /etc/kiwi.yml
    --debug
        print debug information
    -v --version
        show program version
    help
        show manual page

global options for services: image, system
    --logfile=<filename>
        create a log file containing all log information including
        debug information even if this is was not requested by the
        debug switch
    --profile=<name>
        profile name, multiple profiles can be selected by passing
        this option multiple times
    --shared-cache-dir=<directory>
        specify an alternative shared cache directory. The directory
        is shared via bind mount between the build host and image
        root system and contains information about package repositories
        and their cache and meta data.
    --type=<build_type>
        image build type. If not set the default XML specified
        build type will be used

SystemExit: 1
> /home/jiri/.venv/bin/kiwi-ng(34)<module>()
-> sys.exit(load_entry_point('kiwi==9.23.20', 'console_scripts', 'kiwi-ng')())
(Pdb) sys.argv
['/home/jiri/.venv/bin/kiwi-ng', '--type', 'iso', 'system', 'boxbuild', '--box', 'leap', '--description', 'suse-leap-1
5.2', '--target-dir', '/home/jiri/tmp/myimage']
```

Let me know how to get more info if needed. Thanks.

**Comment by schaefi** [comment]
The way you called box build was invalid according to its usage and that's why you get the usage message.
You called it with

```
kiwi-ng --type iso system boxbuild --box leap --description suse-leap-15.2 --target-dir /home/jiri/tmp/myimage
```

and that is invalid because all options are seen as options for the boxbuild command an no ```<kiwi_build_command_args>``` command options (for the call inside of the box were given). I agree with you this can be easily overlooked from the usage message of the positional commandline arguments but we displayed it and also documented it as such in the example:

* https://osinside.github.io/kiwi/self_contained.html#building-with-the-boxbuild-command

From here the correct call in your case would be

```
kiwi-ng --type iso system boxbuild --box leap -- --description suse-leap-15.2 --target-dir /home/jiri/tmp/myimage
```

Let me know if that helps. I would also be interested where in the documentation we made a mistake as you said you used information from the documentation. If you think the docs needs to be improved such that the caller args becomes more clear, let us know about your ideas and we update the docs or come up with a hint/warning box about the special options handling

Thoughts ?
